.github/workflows/preview.yml

```diff
@@ -0,0 +1,76 @@
+name: Amplify Preview
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: gh pr checkout ${{ github.event.issue.number }}
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  conditionalone:
+    if: github.event.issue.pull_request
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: check if PR
+        run: |
+          echo "is PR"
+  conditionaltwo:
+    if: github.event.comment.author_association == 'MEMBER'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: check if PR
+        run: echo "is member"
+  conditionalthree:
+    if: github.event.comment.author_association == 'CONTRIBUTOR'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: check if PR
+        run: echo "is contributor"
+  conditionalfour:
+    needs: prepare
+    if: contains(github.event.comment.body, 'generate preview')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: check if PR
+        run: echo "has comment with 'generate preview' in it"
+  deploy:
+    needs: prepare
+    if: |
+      github.event.issue.pull_request &&
+      (github.event.comment.author_association == 'MEMBER' || github.event.comment.author_association == 'CONTRIBUTOR') &&
```

Comment:
I know we don't work with contractors now, but if we did, would `member` and `contributor` allow a contractor to run this workflow or would we need to add `COLLABORATOR`?

Reply:
Typically, we add contractors as members of the org when giving permissions.